pep-0605.rst

```diff
@@ -0,0 +1,875 @@
+  feature complete for the next X.Y.0 release
+* as with current alpha releases, all APIs added since the last CPython feature
+  release (other than additions to the stable C ABI) would be considered
+  provisional
+* as with current alpha releases, beta releases would be prepared and published
+  from the master development branch
+
+Rather than being released monthly for a period of a few months while preparing
+a new X.Y.0 release, beta releases would instead be consistently published every
+two months.
+
+The only time this would not be the case is during the two month release
+candidate period for an upcoming X.Y.0 release.
+
+
+Changes to release candidate policy, phase duration, and cadence
+----------------------------------------------------------------
+
+Given the proposed changes to the beta release phase, the release candidate
+phase would see the following related adjustments:
+
+* Feature freeze, ABI freeze, pyc file format freeze, and maintenance branch
+  creation would all correspond with the creation of X.Y.0rc1 (currently these
+  occur across a mixture of X.Y.0b1, the last beta release, and X.Y.0rc1)
+* The X.Y.0 release candidate period would be extended from 3 weeks to 2 months
+* There would normally be two release candidates issued a month apart, but
+  additional candidates may be published at the release manager's discretion
+* The final X.Y.0 release would occur between 1 and 4 weeks after the final
+  release candidate (depending if additional release candidates were needed
+  after the second)
+* If the final X.Y.0 release is delayed beyond the August target date, the
+  subsequent release series is not affected, and will still be scheduled for
+  August (now slightly less than two years later).
+
+In addition to allowing more time for end user feedback on the release
+candidate, this adjusted policy also provides additional time for maintainers
+of Python projects to build and publish pre-built wheel archives for the new
+stable release series, significantly improving the initial user experience of
+the X.Y.0 release.
+
+
+Removal of routine alpha releases
+---------------------------------
+
+Due to the proposed policy changes for beta releases and release candidates,
+the traditional CPython alpha release period would no longer serve a clear
+purpose, and as such would be skipped, with the first pre-release in a new
+series being X.Y.0b1.
+
+However, when it seems appropriate to do so, release managers may choose to
+declare that a scheduled beta release will be marked as an alpha release
+instead. For example, a release manager may choose to do this when major changes
+have been made to the release management tooling, when a particularly
+large feature has landed that the release manager views as having a high chance
+of introducing unintended side effects, or when a new interface has been added
+to the stable C ABI that requires further testing and design feedback before
+being locked down in a beta release.
```

Comment:
Is it obvious enough that the stable C ABI "restriction" (permanent as soon as published in beta) can be avoided by just not putting the new APIs into the stable ABI on first release? That seems obvious to me, but perhaps we should call it out explicitly?

Reply:
I would call it out explicitly. I thought it was a bit extreme until I realized that it could go into the CPython API first.

Reply:
The next push adds a parenthetical example that tries to make this clearer (there are some functions that only have purpose if they're part of the stable ABI, so they can't go through the usual CPython ABI -> stable ABI promotion path).